### ISSUE TITLE
CA1812 should not warn when type is used in attribute

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#5658]
### Analyzer

# CA1812: Avoid uninstantiated internal classes

```cs
[DebuggerTypeProxy(typeof(DebugView))]
class C
{
    private sealed class DebugView
    {
    }
}
```

**Comment by Evangelink** [comment]
That's really strange, I did a fix for this issue: #4241

Could you let us know the version of the analyzers you are using?

**Comment by JohanLarsson** [comment]
Everything should be latest at the time of posting

**Comment by Evangelink** [comment]
I have just tested and all unit tests are working as expected + creating a new console app (ensuring the rule `CA1812` is enabled) and pasting your code doesn't produce any issue.

**Comment by Evangelink** [comment]
@JohanLarsson Is it good for you too?

**Comment by JohanLarsson** [comment]
Think there is a bug but if the code in the issue does not repro the issue is not good.
All my versions of Visual Studio are currently broken so can't create a repro.

**Comment by Evangelink** [comment]
Feel free to let us know if you still hit the issue.

**Comment by JohanLarsson** [comment]
Yep, I'll try to produce a proper repro, assumed that code would repro based on what I saw. Thanks for investigating it, sorry about the poor issue.
